sidecar/src/service/sidecar_server.rs

```diff
@@ -0,0 +1,723 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO: APM-1076 - This file contains a fair amount of expects. While in most cases it is unlikely
+// we will ever hit these, we should consider adding more robust error handling.
```

Comment:
I think in these cases the error handling we have is quite fine. In fact, the error handling is tokio catching panics in async tasks and transforming the tasks into an error Result.
The important part is that these panics on expect() don't bring the application into a deadlocked state where e.g. no further traces are processed at all.

On top of that a crash of the whole sidecar, while a minor data loss, is not catastrophic either as it will be restarted by the PHP processes quickly.

Reply:
@bwoebi 
> I'm not a big proponent of having a proper error handling code for things which are not supposed to be possible

I tend to agree as long as 1) we are confident that these errors should never happen and 2) If 1) is wrong, we don't put the application in a bad state. Your comment indicates that we satisfy both these points so I'm ok with leaving these expects as is. I can remove the TODO.